src/Logger.ts

```diff
@@ -0,0 +1,90 @@
+import { LoggerFactory, Level, Event, Factory, NextFunc, Middleware } from './LoggerFactory';
+
+interface LogFunc {
+  (...value: any[]): void;
+}
+
+type LevelKey = Exclude<keyof typeof Logger.prototype, 'log'>;
+
+interface LogLevelFunc {
+  /** Log at provided level. */
+  (level: LevelKey, ...params: any[]): void;
+  /** Log at info level */
+  (...params: any[]): void;
+}
+
+export class Logger {
+  constructor(private name: string, private sink: Factory, private chain: Middleware[], private logLevel: Level) {}
+
+  static getLogger(name = '/'): Logger {
+    return LoggerFactory.getLogger(name);
+  }
+
+  log: LogLevelFunc = (level?: LevelKey, ...params) => {
+    if (level) {
+      const levelKey = this.capitalize(level) as keyof typeof Level;
```

Comment:
If we need to capitalize it to make it a `LevelKey`, then it's not a `LevelKey`. Can we fix this to be more proper with type checking/handling?